packages/terra-pills/src/Pill.jsx

```diff
@@ -0,0 +1,212 @@
+import React, { useRef, useEffect } from 'react';
+import { injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import {
+  KEY_SPACE,
+  KEY_RETURN,
+  KEY_DELETE,
+  KEY_BACK_SPACE,
+} from 'keycode-js';
+import ThemeContext from 'terra-theme-context';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
+import styles from './Pill.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * The label text for the pill. (Required)
+   */
+  label: PropTypes.string.isRequired,
+  /**
+   * The html `id` attribute for the pill.
+   */
+  id: PropTypes.string,
+  /**
+   * A callback function to execute when the pill is removed. If specified, the pill will be removable.
+   */
+  onRemove: PropTypes.func,
+  /**
+   * A callback function to execute when the pill is selected. If specified, the pill label will be selectable.
+   *
+   * ![IMPORTANT](https://badgen.net/badge/UX/Design-Standards/blue) Intended to only be used to disclose a popup.
+   */
+  onSelect: PropTypes.func,
+  /**
+   * Callback to expose pill element's ref for popup placement.
+   */
+  refCallback: PropTypes.func,
+  /**
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) Consumers need to set the `aria-haspopup`
+   * attribute to `true` when using a popup via 'onSelect'. Only applies when used along with the 'onSelect' prop.
+   */
+  ariaHasPopup: PropTypes.bool,
+  /**
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) Consumers need to set the `aria-expanded` attribute to `true`
+   * when using a popup via 'onSelect' and the popup is visible. Only applies when used along with the 'onSelect' prop.
+   */
+  ariaExpanded: PropTypes.bool,
+  /**
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) Consumers need to provide the string of the ID for the
+   * html element containing the popup content, so the `aria-controls` attribute can correctly be associated to the pill.
+   * Only applies when used along with the 'onSelect' prop.
+   */
+  ariaControls: PropTypes.string,
```

Comment:
are we 100% sure we need this? what if the popup isn't rendered on the dom until the on click event? How does this affect screen readers?